### PR TITLE
[docs] Add a note about `AdapterDateFnsV3` on the Getting Started page

### PR DESCRIPTION
**Pull request by flaviendelangle**
Not the prettiest code :grimacing: 
But I think it will reduce the amount of people trying to use `AdapterDateFns` with the latest date-fns releases

### CODE REVIEW

File: docs/data/date-pickers/adapters-locale/adapters-locale.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ For `date-fns`, import the locale and pass it to `LocalizationProvider`:
 :::info
 Both `date-fns` major versions (v2.x and v3.x) are supported.
 
-A single adapter can not work for both `date-fns` v2.x and v3.x, because the way functions are exported has been changed in v3.x.
+A single adapter cannot work for both `date-fns` v2.x and v3.x, because the way functions are exported has been changed in v3.x.
```
Comment (flaviendelangle) [inline docs/data/date-pickers/adapters-locale/adapters-locale.md:46]: https://www.merriam-webster.com/grammar/cannot-vs-can-not-is-there-a-difference#:~:text=Both%20cannot%20and%20can%20not,general%2C%20it%20is%20somewhat%20informal.

Comment (flaviendelangle) [reply docs/data/date-pickers/adapters-locale/adapters-locale.md:46]: I'll make a PR to remove all the "can't" and "can not" from our doc :laughing:

File: docs/src/modules/components/PickersRenderingInstructions.js [hunk 2]
```diff
@@ -21,7 +21,16 @@ export default function PickersRenderingInstructions() {
...
   const commandLines = [
     `import { LocalizationProvider } from '${componentPackage}';`,
+    ...(libraryUsed === 'date-fns'
+      ? ['// If you are using date-fns v2.x, please import `AdapterDateFns`']
+      : []),
     `import { ${adapterName} } from '${componentPackage}/${adapterName}'`,
+    ...(libraryUsed === 'date-fns'
+      ? [
+          '// If you are using date-fns v3.x, please import `AdapterDateFnsV3`',
+          "import { AdapterDateFnsV3 } from '@mui/x-date-pickers/AdapterDateFnsV3'",
```
Comment (LukasTy) [inline docs/src/modules/components/PickersRenderingInstructions.js:31]: Screenshot 2024-02-08 at 10 29 09

```suggestion
          `import { AdapterDateFnsV3 } from '${componentPackage}/AdapterDateFnsV3'`,
```

**Review by LukasTy — COMMENTED** [review]
Nice improvement! 👍

**Review by LukasTy — APPROVED** [review]

**Pull request merged**